pkg/api/handlers/libpod/quadlets.go

```diff
@@ -35,3 +35,25 @@ func ListQuadlets(w http.ResponseWriter, r *http.Request) {
 
 	utils.WriteResponse(w, http.StatusOK, quadlets)
 }
+
+func GetQuadlet(w http.ResponseWriter, r *http.Request) {
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+	name := utils.GetName(r)
+
+	containerEngine := abi.ContainerEngine{Libpod: runtime}
+
+	// Use QuadletPrint to get the raw file contents
+	quadletContents, err := containerEngine.QuadletPrint(r.Context(), name)
+	if err != nil {
+		utils.Error(w, http.StatusNotFound, fmt.Errorf("no such quadlet: %s: %w", name, err))
+		return
+	}
+
+	// Set content type to text/plain and write the raw file contents
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(http.StatusOK)
+	if _, err := w.Write([]byte(quadletContents)); err != nil {
+		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("failed to write quadlet contents: %w", err))
```

Comment:
AFAIK that doesn't work, once it writes 200 OK as status we can no longer return an error because one cannot change a HTTP status mid stream.

And if we failed to write to the content we almost certainly will fail to write the error messge as well, and if that then would work the caller would think the error message is part of the file content.

So in short I would say just log it with logrus on the server